api/src/pcapi/routes/backoffice_v3/fraud.py

```diff
@@ -0,0 +1,159 @@
+from operator import attrgetter
+
+from flask import flash
+from flask import redirect
+from flask import render_template
+from flask import url_for
+from flask_login import current_user
+from flask_sqlalchemy import BaseQuery
+import sqlalchemy as sa
+
+import pcapi.core.history.api as history_api
+import pcapi.core.history.models as history_models
+import pcapi.core.permissions.models as perm_models
+from pcapi.core.users.api import suspend_account
+import pcapi.core.users.constants as users_constants
+import pcapi.core.users.models as users_models
+
+from . import utils
+from .forms import fraud as forms
+
+
+fraud_blueprint = utils.child_backoffice_blueprint(
+    "fraud",
+    __name__,
+    url_prefix="/fraud",
+    permission=perm_models.Permissions.FRAUD_ACTIONS,
+)
+
+
+def render_domain_names_list(form: forms.BlacklistDomainNameForm | None = None) -> utils.BackofficeResponse:
+    if not form:
+        form = forms.PrepareBlacklistDomainNameForm()
+
+    history = (
+        history_models.ActionHistory.query.filter_by(actionType=history_models.ActionType.BLACKLIST_DOMAIN_NAME)
+        .order_by(history_models.ActionHistory.actionDate.desc())
+        .limit(50)
+        .options(
+            sa.orm.joinedload(history_models.ActionHistory.authorUser).load_only(
+                users_models.User.id, users_models.User.firstName, users_models.User.lastName
+            )
+        )
+    )
+
+    return render_template("fraud/domain_names.html", history=history, form=form)
+
+
+@fraud_blueprint.route("", methods=["GET"])
+def list_blacklisted_domain_names() -> utils.BackofficeResponse:
+    return render_domain_names_list()
+
+
+def _filter_beneficiaries_by_domain_name_query(domain_name: str) -> BaseQuery:
+    return users_models.User.query.filter(
+        sa.not_(users_models.User.has_pro_role),
+        sa.not_(users_models.User.has_non_attached_pro_role),
+        users_models.User.email.like(f"%@{domain_name.lower()}"),
+    )
+
+
+def _list_beneficiary_suspensions(domain_name: str) -> list[str]:
+    query = _filter_beneficiaries_by_domain_name_query(domain_name).options(
+        sa.orm.load_only(users_models.User.id, users_models.User.email)
+    )
+
+    return sorted(query, key=attrgetter("email"))
+
+
+def _list_untouched_pro_accounts(domain_name: str) -> list[str]:
+    query = users_models.User.query.filter(
+        sa.or_(  # type: ignore
+            users_models.User.has_pro_role,
+            users_models.User.has_non_attached_pro_role,
+        ),
+        sa.or_(False, users_models.User.email.like(f"%@{domain_name.lower()}")),  # type: ignore
```

Comment:
```suggestion
        users_models.User.email.like(f"%@{domain_name.lower()}"),
```
Déjà suggéré dans la revue précédente : `False or expression` est équivalent à `expression`